test/E2ETest/Infrastructure/EqualityYamlNodeVisitor.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using Tye.Serialization;
+using YamlDotNet.RepresentationModel;
+
+namespace E2ETest
+{
+    public class EqualityYamlNodeVisitor
+    {
+        public EqualityYamlNodeVisitor()
+        {
+        }
+
+        public void Visit(YamlNode node, YamlNode otherNode)
+        {
+            if (node.NodeType == YamlNodeType.Mapping)
+            {
+                VisitMapping((YamlMappingNode)node, (YamlMappingNode)otherNode);
+            }
+            else if (node.NodeType == YamlNodeType.Scalar)
+            {
+                VisitScalar((YamlScalarNode)node, (YamlScalarNode)otherNode);
+            }
+            else if (node.NodeType == YamlNodeType.Sequence)
+            {
+                VisitSequence((YamlSequenceNode)node, (YamlSequenceNode)otherNode);
+            }
+        }
+
+        public void VisitSequence(YamlSequenceNode node, YamlSequenceNode otherNode)
+        {
+            if (otherNode == null)
+            {
+                throw new TyeYamlException($"Expected sequence to be not null, difference starting at Expected: ({node.Start.Line}, {node.Start.Column})");
```

Comment:
Should we track the context (path to current node) for use in error messages? We could include the current context, and since you'd have that you could also track the parent node of `otherNode` which would be able to give you line number info of both expected and actual.

Reply:
I don't see much reason to track the entire path, but the previous node makes sense.

Reply:
Didn't add a context here. I realized the only case where this could be null is in VisitYamlMapping, which I now handle at the location.